### PR TITLE
Rewrite the submission side code

### LINKED ISSUE #409: Give TAs download option for submitted files (and generated files from autograding?)

### PR DESCRIPTION
**Pull request by MasterOdin**
This rewrites the submission code into proper MVC (and OOP style) which makes the logic separated out into logical units (models) instead of having everything in controller_functions.php or data_functions.php 

Current issues:  
* hardcoded semester 'f16' in TA grading site. Need to add to the JS at the bottom of the page to add semester like I did course for now
* write to/from ini file in TA grading site instead of the config table (currently just reads from the ini file)
* Haven't tested SVN upload at all
* Need to add submission url to the configure script

* Add logout button. 

closes #409, #408, #395, #336, #298, #252 (should not list gradeables without assignment_config and let the system into the autograder),

---
This change is [Reviewable](https://reviewable.io/reviews/submitty/submitty/443)

### CODE REVIEW

File: site/app/views/ErrorView.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      `--'   | |/______________\| |   `--'
+             |____________________|
+              `.________________,'
+               (_______)(_______)
+               (_______)(_______)
+               (_______)(_______)
+               (_______)(_______)
+              |        ||        |
+              '--------''--------'
+</pre>
```
Comment (bmcutler) [inline site/app/views/ErrorView.php]: ?

**Comment by bmcutler** [comment]
I scrolled through it and reviewed the changes to grade_students.sh and a few other select files.
Difficult to review since so much is brand new.  @MasterOdin if you think its mostly functional, let's get it merged and make issues for problems when we find them.

File: site/app/views/ErrorView.php (hunk 1 above)
Comment (MasterOdin) [inline site/app/views/ErrorView.php]: It's the error robot!

robot

This (and the Dilbert forbidden page for accessing a course you're not a member of) were just placeholder's till we had something more styled for it.

**Comment by MasterOdin** [comment]
The grade_students.sh change is contained at https://github.com/Submitty/Submitty/blob/1f5b9ded35c11173581eb1139b1dbe3cbc32b02a/bin/grade_students.sh#L415-L420 (the rest of them is my IDE just auto removing whitespace at the end of a line). 

The other changes outside of the php was to use .ini files instead of the .php files for configuration (and to do away with the config table). 

Things appear functional, but definitely require more testing. The upload classlist as xlsx is broken at the moment, but I want to rewrite how that works functionally so it isn't doing redirect business for it.

**Pull request merged**